src/tests/suite/daos_add_simple.c

```diff
@@ -0,0 +1,503 @@
+			       buf_len, &req);
+		assert_int_equal(number, 1);
+
+		number = 10;
+		memset(&anchor, 0, sizeof(anchor));
+		enumerate_akey(DAOS_TX_NONE, "dkey", &number, kds, &anchor,
+			       buf, buf_len, &req);
+		assert_int_equal(number, 0);
+	}
+
+	ioreq_fini(&req);
+}
+
+static void
+addition_multiple(void **state)
+{
+	test_arg_t	*arg = *state;
+	daos_obj_id_t	oid;
+	struct ioreq	req;
+	int		i;
+	int		j;
+	int		k;
+
+	if (!test_runable(arg, 4))
+		return;
+
+	oid = dts_oid_gen(OC_RP_3G1, 0, arg->myrank);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+
+	/** Insert 1000 records */
+	print_message("Insert %d kv record in object "DF_OID"\n",
+		      1000, DP_OID(oid));
+	for (i = 0; i < 10; i++) {
+		char	dkey[16];
+
+		sprintf(dkey, "dkey_3_%d", i);
+		for (j = 0; j < 10; j++) {
+			char	akey[16];
+
+			sprintf(akey, "akey_%d", j);
+			for (k = 0; k < 10; k++)
+				insert_single(dkey, akey, k, "data",
+					      strlen("data") + 1,
+					      DAOS_TX_NONE, &req);
+		}
+	}
+	ioreq_fini(&req);
+
+	addition_single_pool_rank(arg, ranks_to_add, 1);
+}
+
+static void
+addition_large_rec(void **state)
+{
+	test_arg_t		*arg = *state;
+	daos_obj_id_t		oid;
+	struct ioreq		req;
+	int			i;
+	char			buffer[5000];
+
+	if (!test_runable(arg, 4))
+		return;
+
+	oid = dts_oid_gen(OC_RP_3G1, 0, arg->myrank);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+
+	/** Insert 1000 records */
+	print_message("Insert %d kv record in object "DF_OID"\n",
+		      KEY_NR, DP_OID(oid));
+	memset(buffer, 'a', 5000);
+	for (i = 0; i < KEY_NR; i++) {
+		char	key[16];
+
+		sprintf(key, "dkey_4_%d", i);
+		insert_single(key, "a_key", 0, buffer, 5000, DAOS_TX_NONE,
+			      &req);
+	}
+	ioreq_fini(&req);
+
+	addition_single_pool_rank(arg, ranks_to_add, 1);
+}
+
+/** create a new pool/container for each test */
+static const struct CMUnitTest addition_tests[] = {
+	{"ADDITION1: addition small rec multiple dkeys",
+	 addition_dkeys, addition_small_sub_setup, test_teardown},
+	{"ADDITION2: addition small rec multiple akeys",
+	 addition_akeys, addition_small_sub_setup, test_teardown},
+	{"addition3: addition small rec multiple indexes",
+	 addition_indexes, addition_small_sub_setup, test_teardown},
+	{"ADDITION4: addition small rec multiple keys/indexes",
+	 addition_multiple, addition_small_sub_setup, test_teardown},
+	{"ADDITION5: addition large rec single index",
+	 addition_large_rec, addition_small_sub_setup, test_teardown},
+	{"ADDITION6: addition records with multiple snapshots",
+	 addition_snap_update_recs, addition_small_sub_setup, test_teardown},
+	{"ADDITION7: addition punch/records with multiple snapshots",
+	 addition_snap_punch_recs, addition_small_sub_setup, test_teardown},
+	{"ADDITION8: addition keys with multiple snapshots",
+	 addition_snap_update_keys, addition_small_sub_setup, test_teardown},
+	{"ADDITION9: addition keys/punch with multiple snapshots",
+	 addition_snap_punch_keys, addition_small_sub_setup, test_teardown},
+};
+
```

Comment:
These tests seems almost same as rebuild_simple() ?  Just curious, how do you make sure the data will be migrated during rank add.